File: java/src/main/java/adapter/ExchangeRateV1Adapter.java
package adapter;

public class ExchangeRateV1Adapter implements ExchangeRateV2 {
    private final ExchangeRateV1 exchangeRateV1;

    public ExchangeRateV1Adapter(ExchangeRateV1 exchangeRateV1) {
        this.exchangeRateV1 = exchangeRateV1;
    }

    @Override
    public double getExchangeRate(Currency from, Currency to) {
        return switch (from) {
            case USD -> exchangeRateV1.fromUSD(to);
            case KRW -> exchangeRateV1.fromKRW(to);
        };
    }
}

File: kotlin/src/main/kotlin/adapter/ExchangeRateV1Adapter.kt
package adapter

class ExchangeRateV1Adapter(private val exchangeRateV1: ExchangeRateV1) : ExchangeRateV2 {
    override fun getExchangeRate(from: Currency, to: Currency): Double {
        return when (from) {
            Currency.USD -> exchangeRateV1.fromUSD(to)
            Currency.KRW -> exchangeRateV1.fromKRW(to)
        }
    }
}